log/log.go

```diff
@@ -30,18 +32,75 @@ var levelPrefix = [...]string{
 	LevelFatal:    "[FATAL] ",
 }
 
+var levelPriority = [...]syslog.Priority{
+	LevelDebug:    syslog.LOG_DEBUG,
+	LevelInfo:     syslog.LOG_INFO,
+	LevelWarning:  syslog.LOG_WARNING,
+	LevelError:    syslog.LOG_ERR,
+	LevelCritical: syslog.LOG_CRIT,
+	LevelFatal:    syslog.LOG_EMERG,
+}
+
 // Level stores the current logging level.
-var Level = LevelDebug
+var Level = LevelInfo
+
+var (
+	// SysLoggers maps each logging level to a *log.Logger that will be used for it.
+	SysLoggers = make([]*golog.Logger, LevelFatal+1)
+	// UseSyslog controls if syslog is used instead of stdout/err.
+	UseSyslog bool
+	// SyslogNetwork controls the network protocol syslog uses.
+	SyslogNetwork string
+	// SyslogRaddr is the address of a remote syslog instance.
+	SyslogRaddr string
+	// SyslogTag is the tag to pass to syslog.
+	SyslogTag string
+)
+
+func init() {
+	flag.IntVar(&Level, "loglevel", LevelInfo, "Log level (0 = DEBUG, 5 = FATAL)")
+	flag.BoolVar(&UseSyslog, "syslog", false, "Whether or not to use syslog logging")
+	flag.StringVar(&SyslogNetwork, "syslog-network", "", "Syslog network to use. Will fail to dial if left blank.")
+	flag.StringVar(&SyslogRaddr, "syslog-remote", "", "Syslog server to use. Defaults to local on empty string.")
+	flag.StringVar(&SyslogTag, "syslog-tag", "", "Syslog tag to use. Will use the name of the executable if left blank.")
+}
+
+func getSysLogger(l int) *golog.Logger {
+	if SysLoggers[l] != nil {
+		return SysLoggers[l]
+	}
+
```

Comment:
Maybe you can add some logic here:

```Go
net := SyslogNetwork
if SyslogRaddr!="" && net=="" {
	net = "udp"
}
w, err := syslog.Dial(net, SyslogRaddr, levelPriority[l], SyslogTag)
```
So that you still get the "default to UDP" behavior, but it still works with local syslog if both fields are left blank.